### PR TITLE
Sentient Disease spawning safeguards

### PR DESCRIPTION
**Pull request by Rhials**
## About The Pull Request

Makes two minor changes to prevent the Sentient Disease antagonist from being totally screwed/invalid upon creation.

The first prevents sentient diseases from being overwritten by other advanced diseases. This should help make them behave a bit more _parallel_ to the automated diseases, without making them outright override them.

The second prevents the Sentient Disease antagonist from rolling whilst less than 75% of the crew is infected with something already or is otherwise "ineligible" for infection.
## Why It's Good For The Game

Prevents sentient disease from getting bulldozed by high transmission viruses. It won't be able to override them or compete with them, but it can still coexist with those diseases and use those bodies as hosts to spread from. 

Avoids sentient disease rolling in situations where it would either A: Not make the current "holy shit everyone is sick medical please fix this" situation any more interesting (unbearable), or B: Die because someone is playing virology and everyone has a hypergigamaxxed happy virus in their body already.

This isn't a buff, so much as it is trying to prevent the moments where Sentient Disease spawns and has so little impact on the round that it remains unperceived until the roundend reveal.
## Changelog
:cl:
balance: Sentient disease will no longer be overridden by other advanced diseases.
balance: Sentient disease will not spawn unless 75% or more of the station are viable hosts for a sentient disease.
/:cl:

### CODE REVIEW

**Comment by Iamgoofball** [comment]
i tried doing the first point and got denied by @Mothblocks so you'll need to convince them

File: code/modules/events/ghost_role/sentient_disease.dm [hunk 1]
```diff
@@ -9,6 +9,26 @@
 	min_wizard_trigger_potency = 4
 	max_wizard_trigger_potency = 7
 
+/datum/round_event_control/sentient_disease/can_spawn_event(players_amt, allow_magic)
+	. = ..()
+	if(!.)
+		return .
+
+	var/host_counter = 0 //Counts how many potential hosts we have
```
Comment (Mothblocks) [inline code/modules/events/ghost_role/sentient_disease.dm:17]: Useless comment, this is just the variable name spelled out

File: code/modules/events/ghost_role/sentient_disease.dm [hunk 2]
```diff
@@ -9,6 +9,26 @@
...
+	for(var/mob/living/carbon/human/candidate in shuffle(GLOB.player_list))
+		if(candidate.stat == DEAD)
+			continue
+		if(!(candidate.mind.assigned_role.job_flags & JOB_CREW_MEMBER)) //We're only checking crew here.
+			continue
+		if(length(candidate.diseases))
+			continue
+		host_counter++
+
+	if(((host_counter / length(GLOB.player_list)) * 100) < 75) //If under 75% of the crew is infected with something or ineligible, maybe it's not the best time to introduce a sentient disease.
```
Comment (Mothblocks) [inline code/modules/events/ghost_role/sentient_disease.dm:29]: This is also a duplicate of the last comment, would rather you remove the previous one

**Review by Mothblocks — CHANGES_REQUESTED** [review]
Would still rather sentient diseases be considered separate from normal diseases in the same way something like monkey fever and such are than to do what you are doing now

**Comment by Mothblocks** [comment]
It's a thing that all the special viruses do where they're separate. Same thing happens with something like appendicitis I think. I don't think it's easy but I don't really like the solution of it not being overridden by advanced diseases as an interim.

**Comment by Mothblocks** [comment]
> Since the sentient disease is just an advanced disease that the user makes decisions for, moving it off of that and into something that would play nicer with virology viruses would probably require some more fundamental changes

Yes I know, I don't expect you to do it in this PR but I'm telling you I don't think ignoring it with advanced diseases is a good idea as an interim step and I do not want it

> Maybe I could do something with bumping the DISEASE_LIMIT up by 1 and/or making it so that SD is the only thing that can infect beyond that limit, instead of the proposed change?

Don't change DISEASE_LIMIT, I just want sentient disease to bypass it just like GBS (I guess not appendicitis) etc. I don't mean in terms of implementation I don't care about that, just in terms of gameplay

**Comment by Fikou** [comment]
> It's sort of fitting that virology can "cure" sentient diseases by creating a stronger virus. It makes sense for virology to be able to, well, interact with and encumber viruses, as opposed to just telling chem the cure (and even then, only if it's stealth). I'd argue it's one of the few ways for virology to actually help crew, aside from just making the usual 5-10 minute starlight noc regen virus and being done.

this is a cure though. you are preventing the disease from spawning at all if you made a good enough virus. people just wont ever see the effects

**Pull request closed without merging**